### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by rtingchaleun**

### CODE REVIEW

**Comment by Ispirett** [comment]
# Reviewing

**Comment by rtingchaleun** [comment]
the **my_count** method seems to work for me.
https://repl.it/@RonaldTingchale/Microverse-mycount-method

Can you give me some hint or direction about what is not working?

**Comment by rtingchaleun** [comment]
the **my_inject** method seems to work for me.
https://repl.it/@RonaldTingchale/Microverse-mycount-method

Can you give me some hint or direction about what is not working?

**Comment by Ispirett** [comment]
> the **my_count** method seems to work for me.
> https://repl.it/@RonaldTingchale/Microverse-mycount-method
> 
> Can you give me some hint or direction about what is not working?

` array.my_count(0) == array.count(3)` Does not give the same answer
` array.my_count == array.count ` You should have a guard case for if no block is given

**Comment by Ispirett** [comment]
> the **my_inject** method seems to work for me.
> https://repl.it/@RonaldTingchale/Microverse-mycount-method
> 
> Can you give me some hint or direction about what is not working?


`array.my_inject(:+) == array.inject(:+) `  or ` tom = proc {|e| e > 3}`
 `array.my_inject(&:tom) == array.inject(&:tom)`  does not work

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    is_none
+  end
+
+  def my_count
+    count = 0
+    my_each do |e|
+      count += 1 if yield(e) == true
+    end
+    count
+  end
```
Comment (maelfosso) [inline main.rb]: Sorry :disappointed: , it's not working
```ruby
irb(main):002:0> ary = [1,2,4,2]
=> [1, 2, 4, 2]
irb(main):003:0> ary.count
=> 4
irb(main):004:0> ary.my_count
Traceback (most recent call last):
        7: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        6: from (irb):4
        5: from main.rb:56:in `my_count'
        4: from main.rb:5:in `my_each'
        3: from main.rb:5:in `each'
        2: from main.rb:6:in `block in my_each'
        1: from main.rb:57:in `block in my_count'
LocalJumpError (no block given (yield))
irb(main):005:0> ary.count(2)
=> 2
irb(main):006:0> ary.my_count(2)
Traceback (most recent call last):
        3: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        2: from (irb):6
        1: from main.rb:54:in `my_count'
ArgumentError (wrong number of arguments (given 1, expected 0))
```

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  def my_all?
+    is_all = true
+    my_each do |e|
+      if yield(e) == false
+        is_all = false
+        return is_all
+      end
+    end
+    is_all
+  end
```
Comment (maelfosso) [inline main.rb]: There is  small bug to fix
```ruby
irb(main):013:0> [nil, true, 99].all?
=> false
irb(main):014:0> [nil, true, 99].my_all?
Traceback (most recent call last):
        7: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        6: from (irb):14
        5: from main.rb:26:in `my_all?'
        4: from main.rb:5:in `my_each'
        3: from main.rb:5:in `each'
        2: from main.rb:6:in `block in my_each'
        1: from main.rb:27:in `block in my_all?'
LocalJumpError (no block given (yield))
```

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+  def is_any?
+    is_any = false
+    my_each { |e|
+      is_any = true if yield(e) == true
+    }
+    is_any
+  end
```
Comment (maelfosso) [inline main.rb]: There is a small bug to fix
```ruby
irb(main):022:0> [nil, true, 99].any?
=> true
irb(main):023:0> [nil, true, 99].is_any?
Traceback (most recent call last):
        7: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        6: from (irb):23
        5: from main.rb:37:in `is_any?'
        4: from main.rb:5:in `my_each'
        3: from main.rb:5:in `each'
        2: from main.rb:6:in `block in my_each'
        1: from main.rb:38:in `block in is_any?'
LocalJumpError (no block given (yield))
```

File: main.rb [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  def my_none?
+    is_none = true
+    my_each do |e|
+      if yield(e) == true
+        is_none = false
+        return is_none
+      end
+    end
+    is_none
+  end
```
Comment (maelfosso) [inline main.rb]: There is a bug 
```ruby
irb(main):028:0> [nil, false].none? 
=> true
irb(main):029:0> [nil, false].my_none? 
Traceback (most recent call last):
        7: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        6: from (irb):29
        5: from main.rb:45:in `my_none?'
        4: from main.rb:5:in `my_each'
        3: from main.rb:5:in `each'
        2: from main.rb:6:in `block in my_each'
        1: from main.rb:46:in `block in my_none?'
LocalJumpError (no block given (yield))
irb(main):030:0> [nil, false, true].none?
=> false
irb(main):031:0> [nil, false, true].my_none?
Traceback (most recent call last):
        7: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        6: from (irb):31
        5: from main.rb:45:in `my_none?'
        4: from main.rb:5:in `my_each'
        3: from main.rb:5:in `each'
        2: from main.rb:6:in `block in my_each'
        1: from main.rb:46:in `block in my_none?'
LocalJumpError (no block given (yield))
irb(main):032:0> 
```

**Comment by bellom** [comment]
Already talk to you on slack, kindly fix the conflits :+1:

**Pull request merged**